cupyx/scipy/stats/_stats_py.py

```diff
@@ -0,0 +1,147 @@
+import warnings
+
+import cupy
+
+
+def _contains_nan(a, nan_policy='propagate'):
+    policies = ['propagate', 'raise', 'omit']
+
+    if nan_policy not in policies:
+        raise ValueError("nan_policy must be one of {%s}" %
+                         ', '.join("'%s'" % s for s in policies))
+
+    try:
+        contains_nan = cupy.isnan(cupy.sum(a))
+    except TypeError:
+        try:
+            contains_nan = cupy.nan in set(a.ravel())
+        except TypeError:
+            contains_nan = False
+            nan_policy = 'omit'
+            warnings.warn("The input array could not be properly "
+                          "checked for nan values. nan values "
+                          "will be ignored.", RuntimeWarning)
+
+    if contains_nan and nan_policy == 'raise':
```

Comment:
We want evaluate `bool(contains_nan)` only when it is truly necessary as it causes device-to-host sync and inccurs the performance. `contains_nan` is a boolean array on GPU.

```suggestion
    if nan_policy == 'raise' and contains_nan:
```